modules/base/rendering/screenspaceinsetblackout.h

```diff
@@ -0,0 +1,161 @@
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2024                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_MODULE_BASE___SCREENSPACEINSETBLACKOUT___H__
+#define __OPENSPACE_MODULE_BASE___SCREENSPACEINSETBLACKOUT___H__
+
+#include <openspace/rendering/screenspacerenderable.h>
+
+#include <openspace/properties/vector/vec2property.h>
+#include <openspace/properties/misc/optionproperty.h>
+
+namespace openspace {
+
+namespace documentation { struct Documentation; }
+
+class ScreenSpaceInsetBlackout : public ScreenSpaceRenderable {
+public:
+    ScreenSpaceInsetBlackout(const ghoul::Dictionary& dictionary);
+
+    bool deinitializeGL() override;
+    void update() override;
+
+    static documentation::Documentation Documentation();
+
+private:
+    class BlackoutShape : public properties::PropertyOwner {
+    public:
+        class PointOwner : public properties::PropertyOwner {
+        public:
+            class Point {
+            public:
+                Point(glm::vec2& inData, std::string identifier, std::string guiName);
```

Comment:
Eek!  there are lots of nested classes here :s  Makes it really difficult to get an overview of the renderable

Consider splitting them out into different files? Would that make sense?

Or could some of them be made into just simple structs and move the update logic to the main class? 

Either way, something needs to be done to simplify this

Reply:
I think splitting it into different files would be more confusing,  but defining them as dumb `struct`s would clean it up a bit

Reply:
Yes that would help a lot